src/types.rs

```diff
@@ -159,6 +195,46 @@ pub struct Block {
     pub mix_hash: Option<String>,
 }
 
+impl ToChecksummed for Block {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(Block {
+            miner: self.miner.to_checksummed().context("miner")?,
+            ..self
+        })
+    }
+}
+
+impl<T: ToChecksummed> ToChecksummed for Option<T> {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(match self {
+            Some(v) => Some(v.to_checksummed()?),
+            None => None,
+        })
+    }
+}
+
+impl ToChecksummed for String {
+    fn to_checksummed(self) -> Result<Self> {
+        Ok(alloy_primitives::Address::from_hex(self)
+            .context("creating address from hex string")?
+            .to_checksum(None))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use alloy_primitives::hex::FromHex;
+
+    #[test]
+    fn alloy_addr() {
+        let checksum =
+            alloy_primitives::Address::from_hex("0xd8da6bf26964af9d7eed9e03e53415d37aa96045")
+                .unwrap()
+                .to_checksum(None);
+        assert_eq!(checksum, "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045");
+    }
+}
+
```

Comment:
Sorry I just realized I left this test mod here that I was experimenting with whiled developing. Will remove or move to the bottom test mod.